src/library/parser.cpp

```diff
@@ -1,36 +1,57 @@
-//
-// C++ Implementation: parser
-//
-// Description: superclass for external formats parsers
-//
-//
-// Author: Ingo Kossyk <kossyki@cs.tu-berlin.de>, (C) 2004
-// Author: Tobias Rafreider trafreider@mixxx.org, (C) 2011
-//
-// Copyright: See COPYING file that comes with this distribution
-//
-//
+#include "library/parser.h"
 
-#include <QtDebug>
 #include <QDir>
 #include <QFile>
 #include <QIODevice>
 #include <QUrl>
+#include <QtDebug>
 
-#include "library/parser.h"
+#include "library/parsercsv.h"
+#include "library/parserm3u.h"
+#include "library/parserpls.h"
 
-Parser::Parser() {
+// static
+bool Parser::isPlaylistFilenameSupported(const QString& playlistFile) {
+    return ParserM3u::isPlaylistFilenameSupported(playlistFile) ||
+            ParserPls::isPlaylistFilenameSupported(playlistFile) ||
+            ParserCsv::isPlaylistFilenameSupported(playlistFile);
 }
 
-Parser::~Parser() {
-}
+// static
+QList<QString> Parser::parseAllLocations(const QString& playlistFile) {
```

Comment:
consider adding ``[[nodiscard]]`` to all of the new pure functions

Reply:
We consider [[nodiscard]] for pure functions as unnecessary verbosity. It is not the business of the function to restrict its usage. 

See: https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#attribute-nodiscard

I think the usage in the standard is a good patter see: 
https://en.cppreference.com/w/cpp/language/attributes/nodiscard

It is used for empty() where it should have been named isEmpty() to distinguished it from clear() and for functions that forwards the ownership like new() and discarding the return value will lead to a memory leak.

Reply:
If you think so...